napari/layers/utils/color_transformations.py

```diff
@@ -66,7 +66,7 @@ def transform_color_with_defaults(
 
 def transform_color_cycle(
     color_cycle: Union[ColorType, cycle], elem_name: str, default: str
-) -> cycle:
+) -> Tuple[cycle, np.ndarray]:
```

Comment:
Maybe instead of `cycle` we should annotate with `Iterable[np.ndarray]`.

Also, it looks like the docstring should be fixed as cycle will return arrays of size 4, not Nx4.

Reply:
This does return a `cycle`, so I'm not sure what the motivation of returning an `Iterable` is? I've updated the docstring.

Reply:
With cycle, we do not have information about the type of elements that we iterate. When using `Iterable[np.ndarray]` we know that each element is `np.ndarray`.

Reply:
It looks like it's possible to annotate `cycle`, so I've done that.

Reply:
![image](https://github.com/napari/napari/assets/3826210/4c39f14b-4080-47b0-9ca6-8374770ce9b1)

Reply:
Ah, my bad, changed to Iterator now!